sktime/forecasting/model_evaluation/_functions.py

```diff
@@ -40,6 +40,69 @@ def _check_strategy(strategy):
         raise ValueError(f"`strategy` must be one of {valid_strategies}")
 
 
+def _check_scores(metrics) -> Dict:
+    """Validate and coerce Metric objects and segregate them based on predict type.
+
+    Parameters
+    ----------
+    metrics : sktime accepted metrics or a list of them or None
+
+    Return
+    ------
+    metrics_type : Dict
+        The key is metric types and its value is a list of its corresponding metrics.
+    """
+    if not isinstance(metrics, List):
+        metrics = [metrics]
+
+    metrics_type = {}
+    for metric in metrics:
+        metric = check_scoring(metric)  # validate and convert to sktime BaseMetric
+        # collect predict type
+        # I think there's no need to check for attribute "get_tag" as all metric class
```

Comment:
probably you're right - I wrote that line of code, I tend to be very defensive in frequently used interfaces, to the extent that it covers conditions that seem like can be only created by intentional tampering by the user with attributes etc.

In this case, I was trying to "defend" against a metric that is implemented using a lower version extension contract for `BaseMetric` descendants, i.e., where a power user who wrote their own metric does not correctly inherit from `BaseMetric`. Note that even now, we do not have a metric extension template yet! So it is not implausible.

Either way, I think we should leave the (over-?) defensive check in there, to avoid that it breaks something for that specific kind of downstream use case.

Reply:
thanks for the clarification. yup agree 100% and that is a very sensible choice to make.